### PR TITLE
 bug(settings): Fix android manage account settings glitch 

### PR DESCRIPTION
**Pull request by dschom**
## Because

- If users sign out in manner where the browser isn't notified, they can end up in a weird state when accessing settings.

## This pull request

- Asks Firefox for the current user
- Checks the current user's session token
- If the current session is token, uses it thereby letting the user access settings
- If the current session is not valid, a message is displayed saying to signout from sync.

## Issue that this pull request solves

Closes: FXA-10029

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

## Other information (Optional)

Edge where this code becomes active can be hit with these two scenarios:
Local cache cleared behind Firefox’s back

1. Create an account email+1@gmail.com
2. Click firefox menu > Sync and save data
3. Select use email instead
4. Sign in with email+1@gmail.com
5. Provide confirmation code
6. Observe settings page
7. Close settings page
8. In new tab (ie not from sync), go to [accounts.stage.mozaws.net/clear](http://accounts.stage.mozaws.net/clear)
9. Click Firefox menu > [tz@mz.com](mailto:tz@mz.com) > Manage account settings
10. Observe that account settings load now
11. Observe that [tz@mz.com](mailto:tz@mz.com) is the active user

Account signed out behind Firefox’s back

1. Create an account [tz@mz.com](mailto:tz@mz.com)
2. Click firefox menu > Sync and save data
3. Sign in
4. Observe settings page
5. Close window
6. In new tab go [accounts.stage.mozaws.net](http://accounts.stage.mozaws.net/clear)/signin
7. Sign in with [tz@mz.com](mailto:tz@mz.com)
8. Click profile image in top right corner > click signout (This told our server to destroy [tz@mozilla.com](mailto:tz@mz.com)’s session!)
9. Close window
10. Click firefox menu > [tz@mz.com](mailto:tz@mz.com) > Manage account settings (you should be logged in still, and this option should be present)
11. Observe Screen that says ‘Session Expired’ and instructs you to to signout from sync**

** Note that in the future we can make this seamless. At the time of writing, the sign out web channel message isn’t getting picked up by Firefox on android though, so to really sign out of sync, the user must do this themselves. A follow has been filed for this.

### CODE REVIEW

File: packages/fxa-content-server/app/scripts/lib/router.js
Comment (LZoog) [inline packages/fxa-content-server/app/scripts/lib/router.js:620]: Why do we need a new page/route for this, can we just create the component and [return it instead of Settings](https://github.com/mozilla/fxa/blob/68b599c5923341bf16282932b3f250c18972b53f/packages/fxa-settings/src/components/App/index.tsx#L240)?

File: packages/fxa-settings/src/components/App/index.test.tsx
Comment (LZoog) [inline packages/fxa-settings/src/components/App/index.test.tsx:277]: typo: resoponse

File: packages/fxa-settings/src/components/App/index.tsx
Comment (LZoog) [inline packages/fxa-settings/src/components/App/index.tsx:107]: typo: appollo

File: packages/fxa-settings/src/components/App/index.tsx
Comment (LZoog) [inline packages/fxa-settings/src/components/App/index.tsx:123]: Should we be re-running the initial metrics query with this session token and if so, could we rely on the response from that query to determine if it's valid, similar to what we're doing now?

File: packages/fxa-settings/src/components/App/index.tsx
Comment (LZoog) [inline packages/fxa-settings/src/components/App/index.tsx:211]: Should we leave some form of this comment around?

File: packages/fxa-settings/src/components/App/index.tsx
Comment (LZoog) [inline packages/fxa-settings/src/components/App/index.tsx:265]: If we do want to leave this as a new page, I think we'd just want to regular navigate.

**Review by LZoog — COMMENTED** [review]
Left some initial comments/questions, thank you again for diving into this.

File: packages/fxa-settings/src/components/App/index.tsx
Comment (LZoog) [inline packages/fxa-settings/src/components/App/index.tsx:94]: Just noting, this will be `true` for any Sync integrations. Previously, this was targeting mobile sync accessing Settings only.

File: packages/fxa-settings/src/pages/SignoutSync/index.tsx
Comment (LZoog) [inline packages/fxa-settings/src/pages/SignoutSync/index.tsx:30]: Hm, we got approval for different copy [here](https://mozilla.slack.com/archives/C4DCECLGJ/p1721340571446629?thread_ts=1721331665.902959&cid=C4DCECLGJ).

File: packages/fxa-settings/src/components/App/index.tsx
Comment (dschom) [reply packages/fxa-settings/src/components/App/index.tsx:123]: Let me double check this... When I was manually testing it seemed like session.isValid, will trigger a rerender when it writes to the cache, and metrics would then be correct. 

That being said, maybe the initial metrics query shouldn't run until this happens? It might just make the code more 'obvious'. I'm happy to take another pass on this, cause it was something that came to mind.

File: packages/fxa-settings/src/components/App/index.tsx
Comment (dschom) [reply packages/fxa-settings/src/components/App/index.tsx:94]: It doesn't seem like mobile should be a special case. After testing this more, it seems like there are different ways one could end the session without FF knowing about it, and despite it being easy to do on android, I don't think it'd be impossible on desktop.

Also, I dropped the check on config.sendFxAStatusOnSettings. This flag seemed liked it muddied the waters as well.

File: packages/fxa-settings/src/components/App/index.tsx
Comment (dschom) [reply packages/fxa-settings/src/components/App/index.tsx:123]: In addition to changing the appollo cache. We setIsSignedIn on line 133 which will also trigger the rerender.

File: packages/fxa-settings/src/components/App/index.tsx
Comment (dschom) [reply packages/fxa-settings/src/components/App/index.tsx:265]: Just curious why does it matter, since there is no reach router state? FWIW, the other redirect, which already existed was a hard navigate, so I just kept it consistent with that. Happy to change it though.

File: packages/fxa-settings/src/components/App/index.tsx
Comment (dschom) [reply packages/fxa-settings/src/components/App/index.tsx:211]: I'll try to incorporate in the main effect.

File: packages/fxa-content-server/app/scripts/lib/router.js
Comment (dschom) [reply packages/fxa-content-server/app/scripts/lib/router.js:620]: I thought about just returning a different component, but decided to do a page for the following reasons:

1. We get a view metric.
2. When it comes to do auto login, or support a login button it's more isolated.
3. IMO, if you aren't logged in, you shouldn't be able to end up on the /settings page. 
4. The messaging has nothing to do with the setting page. It just feels like a very separate thing to me.

Comment (LZoog) [reply packages/fxa-content-server/app/scripts/lib/router.js:620]: 1. We could easily emit a new Glean event in a component though, it doesn't need to be its own page. I think that `usePageViewEvent` function you used is for Amplitude only.
2. I could see this, but I'm also not sure if we want this UX, but I think it would be nice. In that Slack thread I linked in another comment, Mark was asking what cases users can get into this with our fixes. I'm wondering if we won't want to iterate on the approach since it will be so edgecasey.
3. I'd say users inside the browser won't be able to see the URL, but if they do that new tab link out linked in that Slack thread, they will, so point taken.
4. Related to 3, so yeah, point taken.

I'll leave it up to you, was just curious what the rationale was. 🙂 (side note: I don't think putting this behind `/settings/sync_signout` is better since it would be a page in settings not requiring a session token, but dang, it would prevent you from having to do all the content-server route modifications.)

File: packages/fxa-settings/src/components/App/index.tsx
Comment (dschom) [reply packages/fxa-settings/src/components/App/index.tsx:274]: I just tired to changing this to navigate, and then I sometimes encounter a hanging spinner. Not sure why yet, but I've pushed the change for hardNavigate back up, so as not to block any manual testing.

File: packages/fxa-settings/src/components/App/index.tsx
Comment (LZoog) [reply packages/fxa-settings/src/components/App/index.tsx:99]: > It doesn't seem like mobile should be a special case. After testing this more, it seems like there are different ways one could end the session without FF knowing about it, and despite it being easy to do on android, I don't think it'd be impossible on desktop.

Yeah, sounds OK to me, just wanted to call it out. We should be running `requestSignedInUser` on `app-start` and that `mergeBrowserAccount` function, but apparently there is a bug around it right now (bug linked in that #sync Slack thread too).

> Also, I dropped the check on config.sendFxAStatusOnSettings. This flag seemed liked it muddied the waters as well.

👍 Vijay and I meant to remove this and didn't get to it.

File: packages/fxa-settings/src/components/App/index.tsx
Comment (LZoog) [reply packages/fxa-settings/src/components/App/index.tsx:274]: `hardNavigate` should force a full page reload (it just does a `window.location.href = whatever`) and is usually used for navigating away from React, and using regular navigate keeps the user in the SPA.

Ah, I think I see what's going on. You have `<SignoutSync path="/signout_sync" />` under Settings routes, so I think you're making it accessible at `/settings/signout_sync` instead of `signout_sync`.

File: packages/fxa-content-server/app/scripts/lib/router.js
Comment (dschom) [reply packages/fxa-content-server/app/scripts/lib/router.js]: I just ended up going with the component approach. This is an edge case, and you are right it touches way less files.

File: packages/fxa-settings/src/components/App/index.tsx
Comment (dschom) [reply packages/fxa-settings/src/components/App/index.tsx]: I've switched over to the component approach.

File: packages/fxa-settings/src/pages/SignoutSync/index.tsx
Comment (dschom) [reply packages/fxa-settings/src/pages/SignoutSync/index.tsx]: I've updated the copy.

**Review by LZoog — APPROVED** [review]
LGTM, I mostly looked from a code perspective but I tested this in iOS at least and can see the session being shared when it's no longer in local storage. It might be good to test this in the pairing flow locally which I know Vijay's got working, and I know we talked about pairing up so I can see it working locally as well, but this also looks fine to me to get on stage for easier testing.

**Pull request merged**
